thunder/tests/test_dynamo.py

```diff
@@ -29,12 +30,125 @@ def func(x):
 
     # out should have grad_fn and its name should be ThunderFunctionBackward
     assert out.grad_fn is not None
-    assert out.grad_fn.name() == "ThunderFunctionBackward"
+    if not dynamic:
+        # If dynamic, while trying to execute `x + 1`, we fail with
+        # "s0 had an unexpected type <class 'torch.SymInt'>. Supported types are (<class 'int'>, <class 'thunder.core.baseutils.NumberProxyInterface'>)")
+        # as the FakeTensor for `x` has shape with SymInt.
+        assert out.grad_fn.name() == "ThunderFunctionBackward"
 
     # We record the GraphModules that was compiled by ThunderCompiler
-    assert len(backend.thunder_to_gm) == 2
-    thunder_func, gm = list(backend.thunder_to_gm.items())[0]
-    assert isinstance(gm, torch.fx.GraphModule)
+    assert len(backend.subgraph_infos) == 2
 
-    # This shouldn't be empty
-    assert last_traces(thunder_func)
+    subgraph_info = backend.subgraph_infos[0]
+    if dynamic:
+        assert len(subgraph_info.compiled_functions) == 2  # Due to Symint!
+    else:
+        assert len(subgraph_info.compiled_functions) == 1
+    idx = 1 if dynamic else 0
+    compiled_fn_info = subgraph_info.compiled_functions[idx]
+    assert compiled_fn_info.compiler == CompilerType.THUNDER
+    assert last_traces(compiled_fn_info.compiled_fn)
+    assert isinstance(compiled_fn_info.original_graph_module, torch.fx.GraphModule)
+
+
+@instantiate(
+    dtypes=NOTHING,
+    executors=[DynamoThunderExecutor],
+    decorators=(pytest.mark.parametrize("dynamic", (True, False, None), ids=("dynamic", "static", "auto")),),
+)
+def test_basic_splitter(executor, device: str, dtype: dtypes.dtype, dynamic: bool | None):
+    x = torch.ones(2, 2, device=device, dtype=dtype, requires_grad=True)
+
+    backend = ThunderCompiler()
+
+    def func(x):
+        # torch.sinc has automatic fallback registered.
```

Comment:
And what's the expected behavior? Is `sinc` passed to Inductor or Thunder?

Reply:
`sinc` is passed to `inductor` because it has automatic fallback instead of native support.

Reply:
Please add it to the comment here, it's difficult to infer the expected behavior from the current comment that `torch.sinc` is registered through a fallback.